<!DOCTYPE html>
<!-- saved from url=(0074)http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html -->
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">


<meta name="generator" content="quarto-0.9.649">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Stuyck, Olivia M">

<title>DADA2 Processing Script for Type 2 Diabetes association</title>
<style class="anchorjs"></style><style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="./DADA2 Processing Script for Type 2 Diabetes association_files/clipboard.min.js"></script>
<script src="./DADA2 Processing Script for Type 2 Diabetes association_files/quarto.js"></script>
<script src="./DADA2 Processing Script for Type 2 Diabetes association_files/popper.min.js"></script>
<script src="./DADA2 Processing Script for Type 2 Diabetes association_files/tippy.umd.min.js"></script>
<script src="./DADA2 Processing Script for Type 2 Diabetes association_files/anchor.min.js"></script>
<link href="./DADA2 Processing Script for Type 2 Diabetes association_files/tippy.css" rel="stylesheet">
<link href="./DADA2 Processing Script for Type 2 Diabetes association_files/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="./DADA2 Processing Script for Type 2 Diabetes association_files/bootstrap.min.js"></script>
<link href="./DADA2 Processing Script for Type 2 Diabetes association_files/bootstrap-icons.css" rel="stylesheet">
<link href="./DADA2 Processing Script for Type 2 Diabetes association_files/bootstrap.min.css" rel="stylesheet">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">DADA2 Processing Script for Type 2 Diabetes association</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Stuyck, Olivia M </p>
          </div>
  </div>
    
    
  </div>
  

</header>

<section id="objective" class="level2">
<h2 class="anchored" data-anchor-id="objective">Objective<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#objective" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a></h2>
<p>The purpose of this script is to conduct the following:</p>
<ol type="1">
<li><p>Data Import - take the trimmed fastq files</p></li>
<li><p>Data Wrangling / QA</p></li>
<li><p>DADA2</p>
<ol type="1">
<li><p>Error rate estimation</p></li>
<li><p>Dereplication</p></li>
<li><p>ASV Calling</p></li>
<li><p>Read pair merging</p></li>
<li><p>Chimera removal</p></li>
</ol></li>
<li><p>DECIPHER - Taxonomic Classification</p></li>
<li><p>Data Export</p>
<ol type="1">
<li><p>ASV counts</p></li>
<li><p>Taxonomy matrix</p></li>
<li><p>ASV Fasta</p></li>
</ol>
<hr></li>
</ol>
<section id="data-import" class="level3">
<h3 class="anchored" data-anchor-id="data-import">1. Data Import<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#data-import" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a></h3>
<p>Lets start by loading the libraries, setting the working directory, and importing the fastq files output from trimmomatic</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dada2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Rcpp</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.2     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.2     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.1     </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/home/jochum00/lmatz/LM_RawSequences/paired"</span>)</span>
<span id="cb6-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="do">######################################</span></span>
<span id="cb6-4"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="do">### DATA WRANGLING</span></span>
<span id="cb6-5"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co">#import the dataframe with the manifest,tsv and make vectors of the fwd and rev</span></span>
<span id="cb6-6"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="do">#######################################</span></span>
<span id="cb6-7"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb6-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-8"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb6-8" aria-hidden="true" tabindex="-1"></a>df<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">"../manifest.tsv"</span>,<span class="at">header =</span> F,<span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">row.names =</span> <span class="cn">NULL</span>)</span>
<span id="cb6-9"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df)<span class="ot">&lt;-</span><span class="st">"samples"</span></span>
<span id="cb6-10"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb6-10" aria-hidden="true" tabindex="-1"></a>df<span class="ot">&lt;-</span>df<span class="sc">%&gt;%</span><span class="fu">mutate</span>(<span class="at">fwd=</span><span class="fu">paste0</span>(samples,<span class="st">".1.paired.fq"</span>),</span>
<span id="cb6-11"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb6-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">rev=</span><span class="fu">paste0</span>(samples,<span class="st">".2.paired.fq"</span>),</span>
<span id="cb6-12"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb6-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">filt_fwd=</span><span class="fu">paste0</span>(samples,<span class="st">".1.paired.filt.fq"</span>),</span>
<span id="cb6-13"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb6-13" aria-hidden="true" tabindex="-1"></a>                <span class="at">filt_rev=</span><span class="fu">paste0</span>(samples,<span class="st">".2.paired.filt.fq"</span>),</span>
<span id="cb6-14"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb6-14" aria-hidden="true" tabindex="-1"></a>                )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<hr>
</section>
</section>
<section id="data-wrangling-qa-filter-and-trimming" class="level2">
<h2 class="anchored" data-anchor-id="data-wrangling-qa-filter-and-trimming">2. Data Wrangling / QA / Filter and trimming<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#data-wrangling-qa-filter-and-trimming" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a></h2>
<p>Make a vector containing the forward, reverse, filtered forward, and filtered reverse read name</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># one holding the file names of all the forward reads</span></span>
<span id="cb7-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb7-2" aria-hidden="true" tabindex="-1"></a>forward_reads <span class="ot">&lt;-</span> df<span class="sc">$</span>fwd</span>
<span id="cb7-3"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co"># and one with the reverse</span></span>
<span id="cb7-4"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb7-4" aria-hidden="true" tabindex="-1"></a>reverse_reads <span class="ot">&lt;-</span> df<span class="sc">$</span>rev</span>
<span id="cb7-5"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co"># and variables holding file names for the forward and reverse</span></span>
<span id="cb7-7"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co"># filtered reads we're going to generate below</span></span>
<span id="cb7-8"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb7-8" aria-hidden="true" tabindex="-1"></a>filtered_forward_reads <span class="ot">&lt;-</span>df<span class="sc">$</span>filt_fwd</span>
<span id="cb7-9"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb7-9" aria-hidden="true" tabindex="-1"></a>filtered_reverse_reads <span class="ot">&lt;-</span> df<span class="sc">$</span>filt_rev</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Lets take a look at the quality profiles of the trimmed reads for the first 5 samples</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQualityProfile</span>(reverse_reads[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The `&lt;scale&gt;` argument of `guides()` cannot be `FALSE`. Use "none" instead as
of ggplot2 3.3.4.
ℹ The deprecated feature was likely used in the dada2 package.
  Please report the issue at &lt;https://github.com/benjjneb/dada2/issues&gt;.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="./DADA2 Processing Script for Type 2 Diabetes association_files/unnamed-chunk-3-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQualityProfile</span>(forward_reads[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="./DADA2 Processing Script for Type 2 Diabetes association_files/unnamed-chunk-3-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Ok looks like the read quality drops off around 200bp for the reverse, so lets truncate reverse reads starting at 200bp for quality thresholds less than Q20, and lets remove reads that are less than 50b[ in length</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb11-1" aria-hidden="true" tabindex="-1"></a>filtered_out <span class="ot">&lt;-</span> <span class="fu">filterAndTrim</span>(forward_reads, </span>
<span id="cb11-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb11-2" aria-hidden="true" tabindex="-1"></a>                              filtered_forward_reads,</span>
<span id="cb11-3"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb11-3" aria-hidden="true" tabindex="-1"></a>                              reverse_reads, </span>
<span id="cb11-4"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb11-4" aria-hidden="true" tabindex="-1"></a>                              filtered_reverse_reads, </span>
<span id="cb11-5"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb11-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">maxEE=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>),</span>
<span id="cb11-6"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb11-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">rm.phix=</span><span class="cn">TRUE</span>, </span>
<span id="cb11-7"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb11-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">multithread =</span> T,</span>
<span id="cb11-8"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb11-8" aria-hidden="true" tabindex="-1"></a>                              <span class="at">minLen=</span><span class="dv">50</span>,</span>
<span id="cb11-9"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb11-9" aria-hidden="true" tabindex="-1"></a>                              <span class="at">truncLen=</span><span class="fu">c</span>(<span class="dv">250</span>,<span class="dv">200</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Ok lets take a look at what that did to our dataset</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(filtered_out) <span class="co"># matrix</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "matrix" "array" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(filtered_out) <span class="co"># 20 2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 12  2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb16-1" aria-hidden="true" tabindex="-1"></a>filtered_out</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                               reads.in reads.out
Buffington_201_001.1.paired.fq    17878     15287
Buffington_201_002.1.paired.fq    19088     17182
Buffington_201_003.1.paired.fq    18775     16559
Buffington_201_005.1.paired.fq    22986     19944
Buffington_201_006.1.paired.fq    19611     17239
Buffington_201_007.1.paired.fq    23630     20788
Buffington_201_008.1.paired.fq    19337     17473
Buffington_201_009.1.paired.fq    13076     11231
Buffington_201_010.1.paired.fq    22531     19794
Buffington_201_011.1.paired.fq    19323     15999
Buffington_201_013.1.paired.fq    17213     15100
Buffington_201_014.1.paired.fq    21178     17680</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co">#           </span></span>
<span id="cb18-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="co"># reads.in reads.out</span></span>
<span id="cb18-3"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Buffington_201_001.1.paired.fq    17878     15287</span></span>
<span id="cb18-4"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Buffington_201_002.1.paired.fq    19088     17182</span></span>
<span id="cb18-5"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Buffington_201_003.1.paired.fq    18775     16559</span></span>
<span id="cb18-6"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Buffington_201_005.1.paired.fq    22986     19944</span></span>
<span id="cb18-7"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Buffington_201_006.1.paired.fq    19611     17239</span></span>
<span id="cb18-8"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Buffington_201_007.1.paired.fq    23630     20788</span></span>
<span id="cb18-9"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Buffington_201_008.1.paired.fq    19337     17473</span></span>
<span id="cb18-10"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb18-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Buffington_201_009.1.paired.fq    13076     11231</span></span>
<span id="cb18-11"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb18-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Buffington_201_010.1.paired.fq    22531     19794</span></span>
<span id="cb18-12"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb18-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Buffington_201_011.1.paired.fq    19323     15999</span></span>
<span id="cb18-13"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb18-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Buffington_201_013.1.paired.fq    17213     15100</span></span>
<span id="cb18-14"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb18-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Buffington_201_014.1.paired.fq    21178     17680</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Not too bad, we didnt loose too many reads so lets keep going</p>
<hr>
</section>
<section id="dada2" class="level2">
<h2 class="anchored" data-anchor-id="dada2">3. DADA2<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#dada2" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a></h2>
<section id="error-rate-estimation" class="level3">
<h3 class="anchored" data-anchor-id="error-rate-estimation">3.1 Error rate estimation<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#error-rate-estimation" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a></h3>
<p>Learn the error rates</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># learn the error rates</span></span>
<span id="cb19-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb19-2" aria-hidden="true" tabindex="-1"></a>err_forward_reads <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(filtered_forward_reads, <span class="at">multithread=</span><span class="cn">TRUE</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>51069000 total bases in 204276 reads from 12 samples will be used for learning the error rates.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 51069000 total bases in 204276 reads from 12 samples will be used for learning the error rates.</span></span>
<span id="cb21-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb21-2" aria-hidden="true" tabindex="-1"></a>err_reverse_reads <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(filtered_reverse_reads, <span class="at">multithread=</span><span class="cn">TRUE</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>40855200 total bases in 204276 reads from 12 samples will be used for learning the error rates.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 40855200 total bases in 204276 reads from 12 samples will be used for learning the error rates.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot the learned error rates</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(err_forward_reads, <span class="at">nominalQ=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis
Transformation introduced infinite values in continuous y-axis</code></pre>
</div>
<div class="cell-output-display">
<p><img src="./DADA2 Processing Script for Type 2 Diabetes association_files/unnamed-chunk-7-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(err_reverse_reads, <span class="at">nominalQ=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis
Transformation introduced infinite values in continuous y-axis</code></pre>
</div>
<div class="cell-output-display">
<p><img src="./DADA2 Processing Script for Type 2 Diabetes association_files/unnamed-chunk-7-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="dereplication" class="level3">
<h3 class="anchored" data-anchor-id="dereplication">3.2 Dereplication<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#dereplication" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a></h3>
<p>dereplicate the forward and reverse reads and give them identical names for matching</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb28-1" aria-hidden="true" tabindex="-1"></a>derep_forward <span class="ot">&lt;-</span> <span class="fu">derepFastq</span>(filtered_forward_reads, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_001.1.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 4602 unique sequences from 15287 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_002.1.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 3557 unique sequences from 17182 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_003.1.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 5143 unique sequences from 16559 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_005.1.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 5919 unique sequences from 19944 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_006.1.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 4054 unique sequences from 17239 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_007.1.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 5756 unique sequences from 20788 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_008.1.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 3211 unique sequences from 17473 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_009.1.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 3771 unique sequences from 11231 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_010.1.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 5630 unique sequences from 19794 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_011.1.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 4498 unique sequences from 15999 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_013.1.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 3422 unique sequences from 15100 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_014.1.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 4404 unique sequences from 17680 total sequences read.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(derep_forward) <span class="ot">&lt;-</span>df<span class="sc">$</span>samples <span class="co"># the sample names in these objects are initially the file names of the samples, this sets them to the sample names for the rest of the workflow</span></span>
<span id="cb53-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb53-2" aria-hidden="true" tabindex="-1"></a>derep_reverse <span class="ot">&lt;-</span> <span class="fu">derepFastq</span>(filtered_reverse_reads, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_001.2.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 4534 unique sequences from 15287 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_002.2.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 3064 unique sequences from 17182 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_003.2.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 4290 unique sequences from 16559 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_005.2.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 5010 unique sequences from 19944 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_006.2.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 4532 unique sequences from 17239 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_007.2.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 4292 unique sequences from 20788 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_008.2.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 3827 unique sequences from 17473 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_009.2.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 3780 unique sequences from 11231 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_010.2.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 5462 unique sequences from 19794 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_011.2.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 4286 unique sequences from 15999 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_013.2.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 3668 unique sequences from 15100 total sequences read.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Dereplicating sequence entries in Fastq file: Buffington_201_014.2.paired.filt.fq</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Encountered 4328 unique sequences from 17680 total sequences read.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(derep_reverse) <span class="ot">&lt;-</span> df<span class="sc">$</span>samples</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="asv-calling" class="level3">
<h3 class="anchored" data-anchor-id="asv-calling">3.3 ASV Calling<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#asv-calling" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a></h3>
<p>Run the DADA2 algorithm to generate the ASVS</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="co">#run DADA2 </span></span>
<span id="cb79-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb79-2" aria-hidden="true" tabindex="-1"></a>dada_forward <span class="ot">&lt;-</span> <span class="fu">dada</span>(derep_forward, <span class="at">err=</span>err_forward_reads, <span class="at">pool=</span>T, <span class="at">multithread=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>12 samples were pooled: 204276 reads in 46370 unique sequences.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 12 samples were pooled: 204276 reads in 46370 unique sequences.</span></span>
<span id="cb81-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb81-2" aria-hidden="true" tabindex="-1"></a>dada_reverse <span class="ot">&lt;-</span> <span class="fu">dada</span>(derep_reverse, <span class="at">err=</span>err_reverse_reads, <span class="at">pool=</span>T, <span class="at">multithread=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>12 samples were pooled: 204276 reads in 44362 unique sequences.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 12 samples were pooled: 204276 reads in 44362 unique sequences.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="read-pair-merging" class="level3">
<h3 class="anchored" data-anchor-id="read-pair-merging">3.4 Read pair merging<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#read-pair-merging" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a></h3>
<p>merging forward and reverse reads</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Merging forward and reverse reads</span></span>
<span id="cb84-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb84-2" aria-hidden="true" tabindex="-1"></a>merged_amplicons <span class="ot">&lt;-</span> <span class="fu">mergePairs</span>(dada_forward, derep_forward, dada_reverse,</span>
<span id="cb84-3"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb84-3" aria-hidden="true" tabindex="-1"></a>                               derep_reverse, <span class="at">trimOverhang=</span><span class="cn">TRUE</span>, <span class="at">minOverlap=</span><span class="dv">150</span>)</span>
<span id="cb84-4"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb84-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-5"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb84-5" aria-hidden="true" tabindex="-1"></a><span class="co"># this object holds a lot of information that may be the first place you'd want to look if you want to start poking under the hood</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Generate a count table</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb85-1" aria-hidden="true" tabindex="-1"></a>seqtab <span class="ot">&lt;-</span> <span class="fu">makeSequenceTable</span>(merged_amplicons)</span>
<span id="cb85-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb85-2" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(seqtab) <span class="co"># matrix</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "matrix" "array" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab) <span class="co">#  12 729</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  12 729</code></pre>
</div>
</div>
</section>
<section id="chimera-removal" class="level3">
<h3 class="anchored" data-anchor-id="chimera-removal">3.5 Chimera removal<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#chimera-removal" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a></h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Chimera identification</span></span>
<span id="cb89-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb89-2" aria-hidden="true" tabindex="-1"></a>seqtab.nochim <span class="ot">&lt;-</span> <span class="fu">removeBimeraDenovo</span>(seqtab, <span class="at">verbose=</span>T) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Identified 195 bimeras out of 729 input sequences.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Identified 195 bimeras out of 729 input sequences.</span></span>
<span id="cb91-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb91-2" aria-hidden="true" tabindex="-1"></a><span class="co"># though we only lost 17 sequences, we don't know if they held a lot in terms of abundance, this is one quick way to look at that</span></span>
<span id="cb91-3"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb91-3" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(seqtab.nochim)<span class="sc">/</span><span class="fu">sum</span>(seqtab) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.9485947</code></pre>
</div>
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="co">#  0.9485947</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Generate an overview of the SV counts throughout all the samples</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Overview of counts throughout</span></span>
<span id="cb94-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb94-2" aria-hidden="true" tabindex="-1"></a><span class="co"># set a little function</span></span>
<span id="cb94-3"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb94-3" aria-hidden="true" tabindex="-1"></a>getN <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">sum</span>(<span class="fu">getUniques</span>(x))</span>
<span id="cb94-4"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb94-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-5"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb94-5" aria-hidden="true" tabindex="-1"></a><span class="co"># making a little table</span></span>
<span id="cb94-6"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb94-6" aria-hidden="true" tabindex="-1"></a>summary_tab <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">row.names=</span>df<span class="sc">$</span>samples, </span>
<span id="cb94-7"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb94-7" aria-hidden="true" tabindex="-1"></a>                          <span class="at">dada2_input=</span>filtered_out[,<span class="dv">1</span>],</span>
<span id="cb94-8"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb94-8" aria-hidden="true" tabindex="-1"></a>                          <span class="at">filtered=</span>filtered_out[,<span class="dv">2</span>], </span>
<span id="cb94-9"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb94-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">dada_f=</span><span class="fu">sapply</span>(dada_forward, getN),</span>
<span id="cb94-10"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb94-10" aria-hidden="true" tabindex="-1"></a>                          <span class="at">dada_r=</span><span class="fu">sapply</span>(dada_reverse, getN),</span>
<span id="cb94-11"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb94-11" aria-hidden="true" tabindex="-1"></a>                          <span class="at">merged=</span><span class="fu">sapply</span>(merged_amplicons, getN),</span>
<span id="cb94-12"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb94-12" aria-hidden="true" tabindex="-1"></a>                          <span class="at">nonchim=</span><span class="fu">rowSums</span>(seqtab.nochim),</span>
<span id="cb94-13"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb94-13" aria-hidden="true" tabindex="-1"></a>                          <span class="at">final_perc_reads_retained=</span><span class="fu">round</span>(<span class="fu">rowSums</span>(seqtab.nochim)<span class="sc">/</span>filtered_out[,<span class="dv">1</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>lets write the summary table to a file for reference</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(summary_tab, <span class="st">"read-count-tracking.tsv"</span>, <span class="at">quote=</span><span class="cn">FALSE</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">col.names=</span><span class="cn">NA</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<hr>
</section>
</section>
<section id="decipher---taxonomic-classification" class="level2">
<h2 class="anchored" data-anchor-id="decipher---taxonomic-classification">4. DECIPHER - Taxonomic Classification<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#decipher---taxonomic-classification" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a></h2>
<p>Lets start by downloading the silva138 reference database (if you dont already have it) and loading it into our environment</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="do">## loading DECIPHER</span></span>
<span id="cb96-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb96-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DECIPHER)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biostrings</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:lubridate':

    intersect, setdiff, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    combine, intersect, setdiff, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,
    table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: S4Vectors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: stats4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'S4Vectors'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:lubridate':

    second, second&lt;-</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    first, rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    expand</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    expand.grid, I, unname</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: IRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'IRanges'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:lubridate':

    %within%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    collapse, desc, slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    reduce</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: XVector</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'XVector'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    compact</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomeInfoDb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Biostrings'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    strsplit</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: RSQLite</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: parallel</code></pre>
</div>
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="co"># download the reference taxonomy data</span></span>
<span id="cb124-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb124-2" aria-hidden="true" tabindex="-1"></a><span class="do">## downloading DECIPHER-formatted SILVA v138 reference</span></span>
<span id="cb124-3"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb124-3" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url=</span><span class="st">"http://www2.decipher.codes/Classification/TrainingSets/SILVA_SSU_r138_2019.RData"</span>, <span class="at">destfile=</span><span class="st">"SILVA_SSU_r138_2019.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in download.file(url =
"http://www2.decipher.codes/Classification/TrainingSets/SILVA_SSU_r138_2019.RData",
: URL
http://www2.decipher.codes/Classification/TrainingSets/SILVA_SSU_r138_2019.RData:
cannot open destfile 'SILVA_SSU_r138_2019.RData', reason 'Permission denied'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in download.file(url =
"http://www2.decipher.codes/Classification/TrainingSets/SILVA_SSU_r138_2019.RData",
: download had nonzero exit status</code></pre>
</div>
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="do">## loading reference taxonomy object</span></span>
<span id="cb127-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb127-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"SILVA_SSU_r138_2019.RData"</span>)</span>
<span id="cb127-3"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb127-3" aria-hidden="true" tabindex="-1"></a><span class="do">## creating DNAStringSet object of our ASVs</span></span>
<span id="cb127-4"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb127-4" aria-hidden="true" tabindex="-1"></a>dna <span class="ot">&lt;-</span> <span class="fu">DNAStringSet</span>(<span class="fu">getSequences</span>(seqtab.nochim))</span>
<span id="cb127-5"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb127-5" aria-hidden="true" tabindex="-1"></a><span class="do">## and classifying</span></span>
<span id="cb127-6"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb127-6" aria-hidden="true" tabindex="-1"></a>tax_info <span class="ot">&lt;-</span> <span class="fu">IdTaxa</span>(<span class="at">test=</span>dna, <span class="at">trainingSet=</span>trainingSet, <span class="at">strand=</span><span class="st">"both"</span>, <span class="at">processors=</span><span class="dv">48</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>================================================================================

Time difference of 271.17 secs</code></pre>
</div>
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################################</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now lets make a DNAStringSet object using the nonchimeric sequences that we have generated from out ASVS</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb130-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb130-1" aria-hidden="true" tabindex="-1"></a><span class="do">## creating DNAStringSet object of our ASVs</span></span>
<span id="cb130-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb130-2" aria-hidden="true" tabindex="-1"></a>dna <span class="ot">&lt;-</span> <span class="fu">DNAStringSet</span>(<span class="fu">getSequences</span>(seqtab.nochim))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>And finally lets taxonomically classify our ASVs based on the Silva db using DECIPHER</p>
<p>NOTE: FOR parallel processing, you need to change the number “cores” to the number of processors that the computer running the analysis has</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb131-1" aria-hidden="true" tabindex="-1"></a>tax_info <span class="ot">&lt;-</span> <span class="fu">IdTaxa</span>(<span class="at">test=</span>dna, <span class="at">trainingSet=</span>trainingSet, <span class="at">strand=</span><span class="st">"both"</span>, <span class="at">processors=</span><span class="dv">48</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>================================================================================

Time difference of 264.61 secs</code></pre>
</div>
</div>
<hr>
</section>
<section id="data-export" class="level2">
<h2 class="anchored" data-anchor-id="data-export">5. Data Export<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#data-export" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a></h2>
<p>Lets make our sequence headers more manageable names (ASV_1, ASV_2…)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb133-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Extracting the standard goods from DADA2</span></span>
<span id="cb133-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb133-2" aria-hidden="true" tabindex="-1"></a><span class="co"># giving our seq headers more manageable names (ASV_1, ASV_2...)</span></span>
<span id="cb133-3"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb133-3" aria-hidden="true" tabindex="-1"></a>asv_seqs <span class="ot">&lt;-</span> <span class="fu">colnames</span>(seqtab.nochim)</span>
<span id="cb133-4"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb133-4" aria-hidden="true" tabindex="-1"></a>asv_headers <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="fu">dim</span>(seqtab.nochim)[<span class="dv">2</span>], <span class="at">mode=</span><span class="st">"character"</span>)</span>
<span id="cb133-5"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb133-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-6"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb133-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(seqtab.nochim)[<span class="dv">2</span>]) {</span>
<span id="cb133-7"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb133-7" aria-hidden="true" tabindex="-1"></a>  asv_headers[i] <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"&gt;ASV"</span>, i, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb133-8"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb133-8" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="asv-counts" class="level3">
<h3 class="anchored" data-anchor-id="asv-counts">5.1 ASV Counts<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#asv-counts" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a></h3>
<p>write out the count matrix to a file</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb134"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb134-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb134-1" aria-hidden="true" tabindex="-1"></a><span class="co"># making and writing out a fasta of our final ASV seqs:</span></span>
<span id="cb134-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb134-2" aria-hidden="true" tabindex="-1"></a>asv_fasta <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rbind</span>(asv_headers, asv_seqs))</span>
<span id="cb134-3"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb134-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(asv_fasta, <span class="st">"ASVs.fa"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="taxonomy" class="level3">
<h3 class="anchored" data-anchor-id="taxonomy">5.2 Taxonomy<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#taxonomy" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a></h3>
<p>Make rank names for the taxonomy data, wrangle out the &gt; sign for the ASV headers, and export the taxonomy data</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb135"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb135-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb135-1" aria-hidden="true" tabindex="-1"></a>ranks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"domain"</span>, <span class="st">"phylum"</span>, <span class="st">"class"</span>, <span class="st">"order"</span>, <span class="st">"family"</span>, <span class="st">"genus"</span>, <span class="st">"species"</span>)</span>
<span id="cb135-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb135-2" aria-hidden="true" tabindex="-1"></a>asv_tax <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">sapply</span>(tax_info, <span class="cf">function</span>(x) {</span>
<span id="cb135-3"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb135-3" aria-hidden="true" tabindex="-1"></a>  m <span class="ot">&lt;-</span> <span class="fu">match</span>(ranks, x<span class="sc">$</span>rank)</span>
<span id="cb135-4"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb135-4" aria-hidden="true" tabindex="-1"></a>  taxa <span class="ot">&lt;-</span> x<span class="sc">$</span>taxon[m]</span>
<span id="cb135-5"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb135-5" aria-hidden="true" tabindex="-1"></a>  taxa[<span class="fu">startsWith</span>(taxa, <span class="st">"unclassified_"</span>)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb135-6"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb135-6" aria-hidden="true" tabindex="-1"></a>  taxa</span>
<span id="cb135-7"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb135-7" aria-hidden="true" tabindex="-1"></a>}))</span>
<span id="cb135-8"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb135-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(asv_tax) <span class="ot">&lt;-</span> ranks</span>
<span id="cb135-9"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb135-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(asv_tax) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="at">pattern=</span><span class="st">"&gt;"</span>, <span class="at">replacement=</span><span class="st">""</span>, <span class="at">x=</span>asv_headers)</span>
<span id="cb135-10"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb135-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-11"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb135-11" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(asv_tax, <span class="st">"ASVs_taxonomy.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">quote=</span>F, <span class="at">col.names=</span><span class="cn">NA</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="asv-fasta" class="level3">
<h3 class="anchored" data-anchor-id="asv-fasta">5.3 ASV Fasta<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#asv-fasta" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a></h3>
<p>Lets write out our fasta file containing our final ASV seqs</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb136-1" aria-hidden="true" tabindex="-1"></a>asv_fasta <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rbind</span>(asv_headers, asv_seqs))</span>
<span id="cb136-2"><a href="http://10.16.5.38:8787/files/lmatz/LM_RawSequences/paired/lmatz_DADA2.html#cb136-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(asv_fasta, <span class="st">"ASVs.fa"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>