<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.280">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Jochum,Michael D">

<title>Differential pathway abundance comparison</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="pathway_deseq2_comparison_annotated_files/libs/clipboard/clipboard.min.js"></script>
<script src="pathway_deseq2_comparison_annotated_files/libs/quarto-html/quarto.js"></script>
<script src="pathway_deseq2_comparison_annotated_files/libs/quarto-html/popper.min.js"></script>
<script src="pathway_deseq2_comparison_annotated_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="pathway_deseq2_comparison_annotated_files/libs/quarto-html/anchor.min.js"></script>
<link href="pathway_deseq2_comparison_annotated_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="pathway_deseq2_comparison_annotated_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="pathway_deseq2_comparison_annotated_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="pathway_deseq2_comparison_annotated_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="pathway_deseq2_comparison_annotated_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Differential pathway abundance comparison</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Jochum,Michael D </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<section id="objective" class="level2">
<h2 class="anchored" data-anchor-id="objective"><strong>Objective</strong></h2>
<p>The purpose of this analysis is to compare the inferred pathway abundance file outputs from picrust2 and compare the Type 2 diabetes to the control cohorts.</p>
<section id="analysis-tasks" class="level4">
<h4 class="anchored" data-anchor-id="analysis-tasks"><strong>Analysis Tasks</strong></h4>
<ol type="1">
<li>Data Importation</li>
<li>Data Wrangling</li>
<li>Data Filtering</li>
<li>DESEQ2 assay - DESeq2 provides methods to test for differential expression by use of negative binomial generalized linear models; the estimates of dispersion and logarithmic fold changes incorporate data-driven prior distributions.&nbsp;
<ol type="1">
<li><p>make associations amongst the diabetes status metadata with the differential abundant pathways.</p></li>
<li><p>output statistical test tables</p></li>
</ol></li>
<li>Data Visualization
<ol type="1">
<li><p>MA plots</p></li>
<li><p>Heatmaps of significant pathways</p></li>
</ol></li>
</ol>
<section id="data-import" class="level5">
<h5 class="anchored" data-anchor-id="data-import">1. Data Import</h5>
<p>Lets begin by loading our libraries</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'readr' was built under R version 4.2.3</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'tidyverse' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'ggplot2' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'tibble' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'tidyr' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'purrr' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'dplyr' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'stringr' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'forcats' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'lubridate' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.2     ✔ purrr     1.0.1
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.2     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: S4Vectors
Loading required package: stats4
Loading required package: BiocGenerics

Attaching package: 'BiocGenerics'

The following objects are masked from 'package:lubridate':

    intersect, setdiff, union

The following objects are masked from 'package:dplyr':

    combine, intersect, setdiff, union

The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs

The following objects are masked from 'package:base':

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which.max, which.min


Attaching package: 'S4Vectors'

The following objects are masked from 'package:lubridate':

    second, second&lt;-

The following objects are masked from 'package:dplyr':

    first, rename

The following object is masked from 'package:tidyr':

    expand

The following objects are masked from 'package:base':

    expand.grid, I, unname

Loading required package: IRanges

Attaching package: 'IRanges'

The following object is masked from 'package:lubridate':

    %within%

The following objects are masked from 'package:dplyr':

    collapse, desc, slice

The following object is masked from 'package:purrr':

    reduce

The following object is masked from 'package:grDevices':

    windows

Loading required package: GenomicRanges
Loading required package: GenomeInfoDb
Loading required package: SummarizedExperiment
Loading required package: MatrixGenerics
Loading required package: matrixStats</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'matrixStats' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'matrixStats'

The following object is masked from 'package:dplyr':

    count


Attaching package: 'MatrixGenerics'

The following objects are masked from 'package:matrixStats':

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars

Loading required package: Biobase
Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.


Attaching package: 'Biobase'

The following object is masked from 'package:MatrixGenerics':

    rowMedians

The following objects are masked from 'package:matrixStats':

    anyMissing, rowMedians</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'ggpubr' was built under R version 4.2.3</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggsci)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'ggsci' was built under R version 4.2.3</code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'knitr' was built under R version 4.2.3</code></pre>
</div>
</div>
<p>Import the pathway abundance tables and metadata files</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>df<span class="ot">&lt;-</span><span class="fu">read.delim</span>(<span class="st">"picrust2/pathways_out/path_abun_unstrat_descrip.tsv"</span>,<span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">header=</span>T)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"IRB_Human_Metadata.txt"</span>, <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>, <span class="at">trim_ws =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 12 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (3): SampleID, Diabetes_Status, Sex
dbl (2): BMI, Age

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
</section>
<section id="data-wrangling" class="level5">
<h5 class="anchored" data-anchor-id="data-wrangling">2. Data Wrangling</h5>
<ul>
<li>split the imported data into a df without the descriptions
<ul>
<li>transpose this data-frame so the pathway names are the row names</li>
<li>convert the counts into an integer form that DESeq2 can accept.</li>
</ul></li>
<li>make a key that maps the pathway name to the description</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>metacyc_abundance<span class="ot">&lt;-</span>df <span class="sc">%&gt;%</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>description) <span class="sc">%&gt;%</span> </span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"pathway"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="at">.cols=</span><span class="fu">everything</span>(),as.integer))</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>key<span class="ot">&lt;-</span>df <span class="sc">%&gt;%</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pathway,description) <span class="sc">%&gt;%</span></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct_all</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We also need to convert the metadata character class to factor class where the Type 2 diabetes class is the first level so DESeq2 will compare it to the control class instead of the other way around.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>metadata<span class="ot">&lt;-</span>metadata<span class="sc">%&gt;%</span><span class="fu">mutate</span>(<span class="at">Diabetes_Status=</span><span class="fu">factor</span>(Diabetes_Status,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">"T2D"</span>,<span class="st">"HC"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data-filtering" class="level5">
<h5 class="anchored" data-anchor-id="data-filtering">3. Data Filtering</h5>
<p>Lets conduct an independent filter that removes any pathways with a total of less than 10 counts from the analysis</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>metacyc_abundance<span class="ot">&lt;-</span>metacyc_abundance<span class="sc">%&gt;%</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">keep=</span><span class="fu">rowSums</span>(metacyc_abundance)<span class="sc">&gt;</span><span class="dv">10</span>)<span class="sc">%&gt;%</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(keep<span class="sc">==</span><span class="cn">TRUE</span>)<span class="sc">%&gt;%</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>keep)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="deseq2" class="level5">
<h5 class="anchored" data-anchor-id="deseq2">4. DESeq2</h5>
<p>Let import the count matrix and metadata into DESeq2 with a study design being the comparison across diabetes status</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> metacyc_abundance, </span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> metadata, </span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span>Diabetes_Status) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This is an optional alternative way to filter the count matrix with less than 10 reads after import into deseq2 and an alternative way to relevel the metatdata comparion class post DESEQ2 import</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co">#filtering pathways with less than 10</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">assay</span>(dds)) <span class="sc">&gt;=</span> <span class="dv">10</span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[keep,]</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="co">#ensureing that the comparison uses the control cohort as the reference</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>dds<span class="sc">$</span>Diabetes_Status <span class="ot">&lt;-</span> <span class="fu">relevel</span>(dds<span class="sc">$</span>Diabetes_Status, <span class="at">ref =</span> <span class="st">"HC"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Ok lets run DESeq2 with the default parameters for now and see what happens and export the results.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds,<span class="at">parallel =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates: 18 workers</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- note: fitType='parametric', but the dispersion trend was not well captured by the
   function: y = a/x + b, and a local regression fit was automatically substituted.
   specify fitType='local' or 'mean' to avoid this message next time.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates, fitting model and testing: 18 workers</code></pre>
</div>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds,)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>ok lets inspect the significant results by</p>
<ul>
<li>converting the deseq2 results object with the comparisons to a data frame</li>
<li>filtering anything with an adjusted p value of &lt; 0.05</li>
<li>importing the description of each pathway</li>
<li>reporting the output as a table</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>res2<span class="ot">&lt;-</span><span class="fu">data.frame</span>(res)<span class="sc">%&gt;%</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj<span class="sc">&lt;</span><span class="fl">0.05</span>)<span class="sc">%&gt;%</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(log2FoldChange<span class="sc">&gt;</span><span class="fl">2.5</span>)<span class="sc">%&gt;%</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(padj))</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(res2)<span class="co">#15 x6</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 15  6</code></pre>
</div>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co">#lets </span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>res2<span class="ot">&lt;-</span>res2<span class="sc">%&gt;%</span><span class="fu">mutate</span>(<span class="at">pathway=</span><span class="fu">rownames</span>(res2))</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>res2<span class="ot">&lt;-</span><span class="fu">inner_join</span>(key,res2)<span class="sc">%&gt;%</span></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(padj,<span class="fu">desc</span>(log2FoldChange))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(pathway)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(res2,</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>      <span class="at">row.names =</span> T,</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>      <span class="at">align =</span> <span class="st">'c'</span>,</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">caption =</span> <span class="st">"Table X. Results of 21 significant differentially abundant microbial inferred pathways associated with Type 2 Diabetes cohort as compared to control cohort. Pathway abundances were inferred from DADA2 derived ASV count matrices and taxonoimcally classified using DECIPHER against the Silva138 database. A comparison across diabetes status was conducted using DESeq2 with an upstream independent filtering of pathways containing less than 10 detected counts. Reported results have a log2 fold change &gt;2.5 and an adjusted pvalue of &lt;0.05. Significance was corrected for multiple comparisons using Benjamini-Hochberg multiple test correction."</span>,</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>      <span class="at">digits =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<caption>Table X. Results of 21 significant differentially abundant microbial inferred pathways associated with Type 2 Diabetes cohort as compared to control cohort. Pathway abundances were inferred from DADA2 derived ASV count matrices and taxonoimcally classified using DECIPHER against the Silva138 database. A comparison across diabetes status was conducted using DESeq2 with an upstream independent filtering of pathways containing less than 10 detected counts. Reported results have a log2 fold change &gt;2.5 and an adjusted pvalue of &lt;0.05. Significance was corrected for multiple comparisons using Benjamini-Hochberg multiple test correction.</caption>
<colgroup>
<col style="width: 1%">
<col style="width: 20%">
<col style="width: 45%">
<col style="width: 5%">
<col style="width: 9%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 4%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: center;">pathway</th>
<th style="text-align: center;">description</th>
<th style="text-align: center;">baseMean</th>
<th style="text-align: center;">log2FoldChange</th>
<th style="text-align: center;">lfcSE</th>
<th style="text-align: center;">stat</th>
<th style="text-align: center;">pvalue</th>
<th style="text-align: center;">padj</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: center;">THREOCAT-PWY</td>
<td style="text-align: center;">superpathway of L-threonine metabolism</td>
<td style="text-align: center;">39.258</td>
<td style="text-align: center;">7.310</td>
<td style="text-align: center;">1.790</td>
<td style="text-align: center;">4.084</td>
<td style="text-align: center;">0.000</td>
<td style="text-align: center;">0.014</td>
</tr>
<tr class="even">
<td style="text-align: left;">2</td>
<td style="text-align: center;">FAO-PWY</td>
<td style="text-align: center;">fatty acid β-oxidation I</td>
<td style="text-align: center;">327.247</td>
<td style="text-align: center;">2.806</td>
<td style="text-align: center;">0.729</td>
<td style="text-align: center;">3.850</td>
<td style="text-align: center;">0.000</td>
<td style="text-align: center;">0.019</td>
</tr>
<tr class="odd">
<td style="text-align: left;">3</td>
<td style="text-align: center;">P105-PWY</td>
<td style="text-align: center;">TCA cycle IV (2-oxoglutarate decarboxylase)</td>
<td style="text-align: center;">603.650</td>
<td style="text-align: center;">3.686</td>
<td style="text-align: center;">1.013</td>
<td style="text-align: center;">3.638</td>
<td style="text-align: center;">0.000</td>
<td style="text-align: center;">0.029</td>
</tr>
<tr class="even">
<td style="text-align: left;">4</td>
<td style="text-align: center;">TCA-GLYOX-BYPASS</td>
<td style="text-align: center;">superpathway of glyoxylate bypass and TCA</td>
<td style="text-align: center;">263.162</td>
<td style="text-align: center;">3.750</td>
<td style="text-align: center;">1.156</td>
<td style="text-align: center;">3.244</td>
<td style="text-align: center;">0.001</td>
<td style="text-align: center;">0.036</td>
</tr>
<tr class="odd">
<td style="text-align: left;">5</td>
<td style="text-align: center;">GLYCOLYSIS-TCA-GLYOX-BYPASS</td>
<td style="text-align: center;">superpathway of glycolysis, pyruvate dehydrogenase, TCA, and glyoxylate bypass</td>
<td style="text-align: center;">459.304</td>
<td style="text-align: center;">3.561</td>
<td style="text-align: center;">1.103</td>
<td style="text-align: center;">3.230</td>
<td style="text-align: center;">0.001</td>
<td style="text-align: center;">0.036</td>
</tr>
<tr class="even">
<td style="text-align: left;">6</td>
<td style="text-align: center;">P125-PWY</td>
<td style="text-align: center;">superpathway of (R,R)-butanediol biosynthesis</td>
<td style="text-align: center;">105.699</td>
<td style="text-align: center;">2.873</td>
<td style="text-align: center;">0.920</td>
<td style="text-align: center;">3.123</td>
<td style="text-align: center;">0.002</td>
<td style="text-align: center;">0.048</td>
</tr>
<tr class="odd">
<td style="text-align: left;">7</td>
<td style="text-align: center;">PROTOCATECHUATE-ORTHO-CLEAVAGE-PWY</td>
<td style="text-align: center;">protocatechuate degradation II (ortho-cleavage pathway)</td>
<td style="text-align: center;">8.725</td>
<td style="text-align: center;">6.595</td>
<td style="text-align: center;">2.216</td>
<td style="text-align: center;">2.976</td>
<td style="text-align: center;">0.003</td>
<td style="text-align: center;">0.049</td>
</tr>
<tr class="even">
<td style="text-align: left;">8</td>
<td style="text-align: center;">PWY-922</td>
<td style="text-align: center;">mevalonate pathway I</td>
<td style="text-align: center;">157.564</td>
<td style="text-align: center;">3.978</td>
<td style="text-align: center;">1.345</td>
<td style="text-align: center;">2.957</td>
<td style="text-align: center;">0.003</td>
<td style="text-align: center;">0.049</td>
</tr>
<tr class="odd">
<td style="text-align: left;">9</td>
<td style="text-align: center;">AST-PWY</td>
<td style="text-align: center;">L-arginine degradation II (AST pathway)</td>
<td style="text-align: center;">84.519</td>
<td style="text-align: center;">3.919</td>
<td style="text-align: center;">1.316</td>
<td style="text-align: center;">2.977</td>
<td style="text-align: center;">0.003</td>
<td style="text-align: center;">0.049</td>
</tr>
<tr class="even">
<td style="text-align: left;">10</td>
<td style="text-align: center;">PWY-5910</td>
<td style="text-align: center;">superpathway of geranylgeranyldiphosphate biosynthesis I (via mevalonate)</td>
<td style="text-align: center;">215.193</td>
<td style="text-align: center;">3.911</td>
<td style="text-align: center;">1.318</td>
<td style="text-align: center;">2.968</td>
<td style="text-align: center;">0.003</td>
<td style="text-align: center;">0.049</td>
</tr>
<tr class="odd">
<td style="text-align: left;">11</td>
<td style="text-align: center;">ARGDEG-PWY</td>
<td style="text-align: center;">superpathway of L-arginine, putrescine, and 4-aminobutanoate degradation</td>
<td style="text-align: center;">135.384</td>
<td style="text-align: center;">3.887</td>
<td style="text-align: center;">1.304</td>
<td style="text-align: center;">2.982</td>
<td style="text-align: center;">0.003</td>
<td style="text-align: center;">0.049</td>
</tr>
<tr class="even">
<td style="text-align: left;">12</td>
<td style="text-align: center;">ORNARGDEG-PWY</td>
<td style="text-align: center;">superpathway of L-arginine and L-ornithine degradation</td>
<td style="text-align: center;">135.384</td>
<td style="text-align: center;">3.887</td>
<td style="text-align: center;">1.304</td>
<td style="text-align: center;">2.982</td>
<td style="text-align: center;">0.003</td>
<td style="text-align: center;">0.049</td>
</tr>
<tr class="odd">
<td style="text-align: left;">13</td>
<td style="text-align: center;">PWY0-1338</td>
<td style="text-align: center;">polymyxin resistance</td>
<td style="text-align: center;">101.311</td>
<td style="text-align: center;">3.374</td>
<td style="text-align: center;">1.093</td>
<td style="text-align: center;">3.086</td>
<td style="text-align: center;">0.002</td>
<td style="text-align: center;">0.049</td>
</tr>
<tr class="even">
<td style="text-align: left;">14</td>
<td style="text-align: center;">PWY-6629</td>
<td style="text-align: center;">superpathway of L-tryptophan biosynthesis</td>
<td style="text-align: center;">635.108</td>
<td style="text-align: center;">3.319</td>
<td style="text-align: center;">1.125</td>
<td style="text-align: center;">2.951</td>
<td style="text-align: center;">0.003</td>
<td style="text-align: center;">0.049</td>
</tr>
<tr class="odd">
<td style="text-align: left;">15</td>
<td style="text-align: center;">PWY-6396</td>
<td style="text-align: center;">superpathway of 2,3-butanediol biosynthesis</td>
<td style="text-align: center;">88.959</td>
<td style="text-align: center;">2.865</td>
<td style="text-align: center;">0.948</td>
<td style="text-align: center;">3.023</td>
<td style="text-align: center;">0.003</td>
<td style="text-align: center;">0.049</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Lets additionally save the analysis as an rds object and write an output of all the pathway comparisons to file for reference</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(dds,<span class="st">"T2D_deseq2_dataset.RDS"</span>)</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(res,<span class="st">"T2D_deseq2_results.RDS"</span>)</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">data.frame</span>(res),<span class="at">file =</span> <span class="st">"T2D_DESeq2_results_full.tsv"</span>,<span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>,<span class="at">row.names =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>5. Data Visualization</p>
<p>Lets visualize the MA plot</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>MA_plot<span class="ot">&lt;-</span><span class="fu">plotMA</span>(res,</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fl">0.1</span>,</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">"Fig S1. MA plot of Diff. abundant pathways assoc. with Diabetes Status"</span>,</span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">xlab =</span> <span class="st">"Mean of Normalized Counts"</span>,<span class="at">ylab=</span><span class="st">"Log2 Fold Change"</span>,</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">7</span>,<span class="dv">7</span>),</span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">cex=</span><span class="fl">0.75</span>,</span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">colNonSig =</span> <span class="st">"gray10"</span>,</span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">colSig =</span> <span class="st">"firebrick"</span>,</span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">colLine =</span> <span class="st">"grey40"</span>,</span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">returnData =</span> <span class="cn">FALSE</span>,</span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">MLE =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="pathway_deseq2_comparison_annotated_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Lets make a heatmap</p>
<div class="cell">

</div>
</section>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>